chili-epfl/FROG
# Accounts in FROG - functions to create a new user and login with password

This PR introduces 2 functions to create a new 'verified' user with a password and email as well as login to that user. We are no longer using the unique username but the email will now be unique and  we will have a profile object for each user that will have a field for DisplayName which is not unique but will be displayed on the UI. Please note in the Meteor.users collection,  the email field is an array of objects (with 2 fields: address and verified (bool)). Verified is used to indicate whether the email is verified but **has nothing to do** with the concept of a verified user in FROG. 

**Meteor Packages used:**

1. accounts-base 
2. accounts-password

**Background**

- There are 2 cases when you want to create a verified account.
  a) By default every user that goes to the landing page of FROG is logged in as an anonymous user. After using the product the user might decide to make an account with us. So when they fill out their information (email and password), we simply use the same anonymous userId and update it with their information. In this case we pass the userId of the anonymous user as a parameter to the create account function. 
b) If they are not anonymous and they want to create another account, then we create a new user in the users collection. 

**Testing instructions**
1. **Account creation**
For now these functions will be tested from the console. 

_Case 1_:

- Navigate to localhost:3000. Note the userid of the anonymous user. 

- Call `Meteor.call('create.account',Meteor.userId(),'testemail@gmail.com','somepassword',{DisplayName:'UserTester'},(err,succ) => console.log(err,succ))`
- The database should have the same userId noted in step 1, but with the updated email, password, profile information 

_Case 2_:
- Log in with a non anonymous user by using `Meteor.loginWithPassword('someuser@gmail.com', 'somepassword')`. Note the userId. 
- Call `Meteor.call('create.account',null,'testemail2@gmail.com','someotherpassword',{DisplayName:'UserTester2nd'}, (err,succ) => console.log(err,succ))`. We are passing null as the userId since the user is not anonymous and we want to create a new user. (This should work even without passing null since we have a check for anonymous user in the function)
- In the `Metoer.users` collection, there should be a new user with a different userId than before. 

**2.Logging in with password**
- Simply call `Meteor.loginWithPassword('email','password', (err,succ) => console.log(err,succ))` from the console. This function is meant to be used only on the client so it will be used directly on the client when we implement the front-end components . 
- Verify that the user is logged in by checking localStorage or using `Meteor.userId() or Meteor.user().emails[0].address`
